exotica_core/src/kinematic_tree.cpp

```diff
@@ -1008,7 +1083,47 @@ void KinematicTree::SetFloatingBaseLimitsPosXYZEulerZYX(
     }
     for (int i = 0; i < 6; ++i)
     {
+        // Set velocity and acceleration to +inf
         controlled_joints_[i].lock()->joint_limits = {lower[i], upper[i]};
+        controlled_joints_[i].lock()->velocity_limit = {inf};
+        controlled_joints_[i].lock()->acceleration_limit = {inf};
+    }
+
+    UpdateJointLimits();
+}
+
+void KinematicTree::SetPlanarBaseLimitsPosXYEulerZ(
+    const std::vector<double>& lower, const std::vector<double>& upper, const std::vector<double>& velocity, const std::vector<double>& acceleration)
+{
+    bool velocity_acceleration_to_inf = false;
+    if (controlled_base_type_ != BaseType::PLANAR)
+    {
+        ThrowPretty("This is not a planar joint!");
+    }
+    if (velocity.size() == 0 && acceleration.size() == 0)
+    {
+        velocity_acceleration_to_inf = true;
+    }
+    else if (lower.size() != 3 || upper.size() != 3 || velocity.size() != 3 || acceleration.size() != 3)
+    {
+        ThrowPretty("Wrong limit data size!");
+    }
+    else
+    {
+    }
```

Comment:
This logic ignores lower/upper if either vel or acc are empty.
Split this into multiple checks
```
if (lower.size() != 3 || upper.size() != 3) ThrowPretty("Wrong joint limit size!");
if (velocity.size() != 3 && velocity.size() != 0) ThrowPretty("Wrong velocity limit size!");
if (acceleration.size() != 3 && acceleration.size() != 0) ThrowPretty("Wrong acceleration limit size!");
```

Reply:
Ah yes - my mistake, implemented